lib/cog/command/pipeline/executor.ex

```diff
@@ -428,50 +428,38 @@ defmodule Cog.Command.Pipeline.Executor do
 
   # For a specific adapter, render each output, concatenating all
   # results into a single response string
-  defp render_templates(command_output, command, adapter) do
-    rendered_templates = Enum.reduce_while(command_output, [], fn({data, template}, acc) ->
-      try do
-        rendered_template = render_template(command, adapter, template, data)
-        {:cont, [rendered_template | acc]}
-      rescue
-        error ->
+  defp render_templates(adapter, bundle, output) do
+    rendered_templates = Enum.reduce_while(output, [], fn({context, template}, acc) ->
+      case render_template(adapter, bundle, template, context) do
+        {:ok, result} ->
+          {:cont, [result|acc]}
+        {:error, error} ->
           {:halt, {:error, {error, template, adapter}}}
       end
     end)
 
     case rendered_templates do
-      {:error, data} ->
-        {:error, data}
+      {:error, error} ->
+        {:error, error}
       messages ->
-        Enum.reverse(messages)
+        messages
+        |> Enum.reverse
         |> Enum.join("\n")
     end
   end
 
-  # Render a single output
-  defp render_template(command, adapter, nil, context),
-    do: render_template(command, adapter, default_template(context), context)
-  defp render_template(command, adapter, template, context) do
-    # If `TemplateCache.lookup/3` returns nil instead of a function,
-    # we know that the adapter doesn't have a template with the given
-    # name. In this case, we can fall back to no template and run
-    # through render_template again to pick up a default
-    #
-    # This is *NOT* a long-term solution.
-    case TemplateCache.lookup(command.bundle.id, adapter, template) do
-      fun when is_function(fun) ->
-        fun.(context)
-      nil ->
-        Logger.warn("The template `#{template}` was not found for adapter `#{adapter}` in bundle `#{command.bundle.name}`; falling back to the default")
-        render_template(command, adapter, nil, context)
+  defp render_template(adapter, bundle, template, context) do
+    case Template.render(adapter, bundle.id, template, context) do
+      {:ok, output} ->
+        {:ok, output}
+      {:error, :template_not_found} ->
```

Comment:
I don't really consider this as part of the fallback logic. This is a real error case where the template specified could not be found, which I think the executor should be able to react to. In the future I expect this clause to eventually render a "not_found" template where we print an error and return the result in json as we currently do.

Also as a minor aside, this prints the bundle name which isn't something the `Template` module knows about. Woudln't be hard to change but I do like the API surface we currently have.